leakcanary-deobfuscation-gradle-plugin/src/main/java/com/squareup/leakcanary/deobfuscation/LeakCanaryLeakDeobfuscationPlugin.kt

```diff
@@ -77,6 +77,12 @@ class LeakCanaryLeakDeobfuscationPlugin : Plugin<Project> {
         ) ?: findTaskProviderOrNull(
             project,
             "transformClassesAndResourcesWithProguardFor${variant.name.capitalize()}"
+        ) ?: findTaskProviderOrNull(
+            project,
+            "minify${variant.name.capitalize()}WithR8"
+        ) ?: findTaskProviderOrNull(
+            project,
+            "minify${variant.name.capitalize()}WithProguard"
         ) ?: throwMissingMinifiedVariantException()
```

Comment:
This looks fine but I suggest to split it into separate methods. We'll need to change it again when new Variant and Artifact Transform API is available in AGP. So I think it's good to start splitting it now. We could have something like `findMappingGeneratingTaskPreAGP3_6`, `findMappingGeneratingTaskAGP3_6` etc... (I don't like those names, it's just an example to show what I mean) so we can later add a separate method when the API changes again. 
Also I think we can assume that users will use the newest plugin, so I would switch the order of those calls so that we first try to find the task using new AGP task names and then fall back to old AGP task names.

Reply:
I'm merging as is, feel free to follow up with a refactor.